packages/core/src/stream/aisdk/v5/input.ts

```diff
@@ -19,11 +29,40 @@ export class AISDKV5InputStream extends MastraModelInput {
     stream: ReadableStream<LanguageModelV2StreamPart>;
     controller: ReadableStreamDefaultController<ChunkType>;
   }) {
+    // Map numeric IDs to UUIDs for uniqueness across steps.
+    // Workaround for @ai-sdk/anthropic and @ai-sdk/google duplicate IDs bug:
+    // These providers use numeric indices ("0", "1", etc.) that reset per LLM call.
+    // See: https://github.com/mastra-ai/mastra/issues/9909
+    const idMap = new Map<string, string>();
+
     // ReadableStream throws TS errors, if imported not imported. What an annoying thing.
     //@ts-ignore
     for await (const chunk of stream) {
-      const transformedChunk = convertFullStreamChunkToMastra(chunk as StreamPart, { runId });
+      const rawChunk = chunk as StreamPart;
+
+      // Clear ID map on new step so each step gets fresh UUIDs
+      if ((rawChunk as { type: string }).type === 'stream-start') {
+        idMap.clear();
+      }
+
+      const transformedChunk = convertFullStreamChunkToMastra(rawChunk, { runId });
+
       if (transformedChunk) {
+        // Replace numeric IDs with UUIDs for text chunks
+        if (
+          (transformedChunk.type === 'text-start' ||
+            transformedChunk.type === 'text-delta' ||
+            transformedChunk.type === 'text-end') &&
+          transformedChunk.payload?.id &&
+          isNumericId(transformedChunk.payload.id)
+        ) {
+          const originalId = transformedChunk.payload.id;
+          if (!idMap.has(originalId)) {
+            idMap.set(originalId, randomUUID());
+          }
+          transformedChunk.payload.id = idMap.get(originalId)!;
```

Comment:
Can't you pass in an id generator for ai sdk? Maybe we want to use that and fallback to `randomUUID()` if it's not set